website/docs/getting-started/template-walkthrough.md

```diff
@@ -0,0 +1,142 @@
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    console.log(event.message);
+    return;
+  }
+);
+
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+We start by importing Constructs from `aws-cdk-lib` and `functionless`:
+
+```typescript
+import { App, Stack } from "aws-cdk-lib";
+import { Function, StepFunction } from "functionless";
+```
+
+:::note
+
+You can read more about Constructs in the [AWS CDK documentation][cdk-construct].
+
+:::
+
+Functionless exposes AWS CDK compatible Constructs from its top level API.
+In this example, we're using the `Function` and `StepFunction` Constructs which map to the AWS Lambda and AWS Step Functions services.
+
+Next, we initialize `app` which will be the root of our CDK Construct tree:
+
+```typescript
+const app = new App();
+```
+
+CDK models infrastructure as a tree of Constructs with a single instance of `App` at the root.
+To add Constructs to the tree, we pass the parent as the first argument when instantiating a child.
+
+This is exactly what we do next with `Stack`:
+
+```typescript
+const stack = new Stack(app, "MyStack");
+```
+
+`stack` will be the parent of the rest of the Constructs we create.
+In CDK, all Resources belong to a Stack - a Stack is a collection of Resources that are deployed and managed together.
+
+:::note
+
+An `App` may contain many stacks, but that is outside the scope of this document.
+You can read more about [`App`][cdk-app] and [`Stack`][cdk-stack] in the CDK documentation.
+
+:::
+
+Once we have an instance of `stack`, we're ready to attach our first `Function`:
+
+```typescript
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    // This code will run on AWS Lambda
+    console.log(event.message);
+    return;
+  }
+);
+```
+
+This might not be the most interesting code in the world, but it offers us our first glimpse at the magic of Functionless.
+
+If you've used CDK before, you're probably familiar with the seperation it imposes between infrastructure and application code.
+Functionless bridges this gap by letting you write application code inline with the infrastructure it sits atop.
+
+This opens up a world of new possibilities for how code and infrastructure can interact.
+We can see this in action with the `StepFunction` Construct definition:
+
+```typescript
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  // This will run in an AWS Step Function
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+There is a lot to unpack in these three lines.
+
+First off, we're able to call the `sayFunction` Construct we just defined as if it were a standard nodejs `function`.
+The types inferred from arrow function in the `Function` definition above are preserved, giving us type-checking when we call `sayFunction`.
```

Comment:
I worry about the use of "inferred types" here. An idea/feedback below:

> Just like you'd expect, the function signature of `sayFunction` is used to provide type-checking, ensuring that your code is free from a whole classes of user errors such as typos and mis-matched contracts. If you've ever deployed a Step Function, only to find that it is passing the wrong data to a Lambda Function, the value of this is clear.

Perhaps a bit wordy - but some ideas.

Reply:
> The arguments and return type of `say` function match what you/we defined above. Functionless uses type-safety to ensure your resources are used as you expect them to be. If you change your `Function` above to accept a `number` instead of a `string`, then typescript will show an error here, keeping your application safe from stray changes.

Reply:
Tried to incorporate the feedback into a complete rework of the paragraph. Hopefully it reads better now.

Reply:
It does!